Library/MeterGeometry.cpp

```diff
@@ -0,0 +1,370 @@
+	std::vector<std::wstring> tokens = CustomTokenize(string, L",");
+	double rotation = 0;
+	if (tokens.size() > 0)  rotation = ConfigParser::ParseDouble(tokens[0].c_str(), defaultValue);
+	if (tokens.size() > 2) {
+		shape.m_RotationCenter = ParsePoint((tokens[1] + L"," + tokens[2]).c_str(), FLT_MIN);
+		if (shape.m_RotationCenter.x == FLT_MIN) shape.m_RotationCenter.x = (shape.m_Bounds.right - shape.m_Bounds.left) / 2 - shape.m_Bounds.left;
+		if (shape.m_RotationCenter.y == FLT_MIN) shape.m_RotationCenter.y = (shape.m_Bounds.bottom - shape.m_Bounds.top) / 2 - shape.m_Bounds.top;
+	}
+	else {
+		shape.m_RotationCenter.x = (shape.m_Bounds.right - shape.m_Bounds.left) / 2 - shape.m_Bounds.left;
+		shape.m_RotationCenter.y = (shape.m_Bounds.bottom - shape.m_Bounds.top) / 2 - shape.m_Bounds.top;
+	}
+	return rotation;
+}
+
+std::vector<std::wstring> MeterGeometry::CustomTokenize(const std::wstring& str, const std::wstring& delimiters)
+{
+	std::vector<std::wstring> tokens;
+
+	size_t lastPos, pos = 0;
+	do
+	{
+		lastPos = str.find_first_not_of(delimiters, pos);
+		if (lastPos == std::wstring::npos) break;
+
+		pos = str.find_first_of(delimiters, lastPos + 1);
+		std::wstring token = str.substr(lastPos, pos - lastPos);  // len = (pos != std::wstring::npos) ? pos - lastPos : pos
+
+		size_t pos2 = token.find_first_not_of(L" \t\r\n");
+		//If a token fix was needed, then are the pos one token off, just needed to keep it right
+		bool offsett = false;
+		if (pos2 != std::wstring::npos)
+		{
+			//Check wether it's even needed to try combine mathematic expressions
+			if (token.find(L'(') != std::wstring::npos || token.find(L'[') != std::wstring::npos || token.find(L'#') != std::wstring::npos)
+			{
+				//Add the next token while the order of ( and ), and [ and ] are still uneven 
+				while (std::count(token.begin(), token.end(), L'(') != std::count(token.begin(), token.end(), L')') &&
+					std::count(token.begin(), token.end(), L'[') != std::count(token.begin(), token.end(), L']') &&
+					std::count(token.begin(), token.end(), L'#') != std::count(token.begin(), token.end(), L'#') &&
+					pos != std::wstring::npos && lastPos != std::wstring::npos)
+				{
+					lastPos = str.find_first_not_of(delimiters, pos);
+					if (lastPos == std::wstring::npos) break;
+					pos = str.find_first_of(delimiters, lastPos + 1);
+					token += str.at(lastPos - 1) + str.substr(lastPos, pos - lastPos);
+					if (pos == std::wstring::npos) break;
+					++pos;
+					offsett = true;
+				}
+			}
+			//End of addition
+			size_t lastPos2 = token.find_last_not_of(L" \t\r\n");
+			if (pos2 != 0 || lastPos2 != (token.size() - 1))
+			{
+				// Trim white-space
+				token.assign(token, pos2, lastPos2 - pos2 + 1);
+			}
+			tokens.push_back(token);
+		}
+
+		if (pos == std::wstring::npos) break;
+		if (!offsett)
+			++pos;
+	} while (true);
+
+	return tokens;
+}
+//Modified version of ParseRect
+D2D1_POINT_2F MeterGeometry::ParsePoint(const LPCWSTR& string, double defaultVal)
```

Comment:
This is probably a better idea, i just have to make the static ParseInt4 function first ;)